src/replication.c

```diff
@@ -2666,6 +2667,12 @@ void syncWithMaster(connection *conn) {
     /* Receive REPLCONF listening-port reply. */
     if (server.repl_state == REPL_STATE_RECEIVE_PORT_REPLY) {
         err = receiveSynchronousResponse(conn);
+        if (!strncmp(err, "-Reading from master", 20)) {
+            serverLog(LL_WARNING, "Master did not respond to "
+                                  "REPLCONF listening-port: %s", err);
+            sdsfree(err);
+            goto error;
+        }
```

Comment:
maybe add another `goto` label for these and handle them in shared code?

Reply:
Sounds good. Will do that